src/metadataGeneration/typeResolver.ts

```diff
@@ -89,8 +89,44 @@ export class TypeResolver {
       return { dataType: 'any' } as Tsoa.Type;
     }
 
-    if (this.typeNode.kind === ts.SyntaxKind.TypeLiteral) {
-      return { dataType: 'any' } as Tsoa.Type;
+    if (ts.isTypeLiteralNode(this.typeNode)) {
+      const properties = this.typeNode.members
+        .filter(member => ts.isPropertySignature(member))
+        .reduce((res, propertySignature: ts.PropertySignature) => {
+          const type = new TypeResolver(propertySignature.type as ts.TypeNode, this.current, propertySignature).resolve();
+
+          return [
+            {
+              default: getJSDocComment(propertySignature, 'default'),
+              description: this.getNodeDescription(propertySignature),
+              format: this.getNodeFormat(propertySignature),
+              name: (propertySignature.name as ts.Identifier).text,
+              required: !propertySignature.questionToken,
+              type,
+              validators: getPropertyValidators(propertySignature),
+            } as Tsoa.Property,
```

Comment:
@WoH can you create a temp variable of type Tsoa.Property instead of using `as`?

So like this instead:
```ts
const aProperty: Tsoa.Property = {
```

Reply:
Done